lib/ui/src/components/preview/preview.js

```diff
@@ -86,20 +86,24 @@ const defaultWrappers = [
 const getTools = memoize(10)(
   (getElements, queryParams, panels, api, options, storyId, viewMode, location, path, baseUrl) => {
     const tools = getElementList(getElements, types.TOOL, [
-      panels.filter(p => p.id !== 'canvas').length
+      panels.filter(p => !p.hidden).length > 1
         ? {
             render: () => (
               <Fragment>
                 <TabBar key="tabs" scroll={false}>
-                  {panels.map((t, index) => {
-                    const to = t.route({ storyId, viewMode, path, location });
-                    const isActive = path === to;
-                    return (
-                      <S.UnstyledLink key={t.id || `l${index}`} to={to}>
-                        <TabButton active={isActive}>{t.title}</TabButton>
-                      </S.UnstyledLink>
-                    );
-                  })}
+                  {panels
+                    .filter(p => !p.hidden)
```

Comment:
nit: could store as const since this code is repeated [here](https://github.com/storybookjs/storybook/pull/9095/files#diff-39c88cb826a0dfeafd9a340db68e59deR89). Obviously the performance isn't a major issue but just in case the requirements to hide the tabs change in the future it would avoid bugs since there would only be one place to update.

Reply:
@liamross I'm merging this. perhaps you'd be able to open a follow-up PR making it just that bit nicer?

Reply:
@ndelangen I don't think it's important enough to actually change, was just a nitpick